GooglePlayInstant/Editor/QuickDeploy/LoadingScreenGenerator.cs

```diff
@@ -233,23 +233,33 @@ private static Vector2 GetPreImportTextureDimensions(Texture2D texture)
             tempTexture.LoadImage(imageBytes);
             return new Vector2(tempTexture.width, tempTexture.height);
         }
-
-        // Visible for testing.
-        internal static Sprite FindReplayButtonSprite()
+        
+        /// <summary>
+        /// Searches the AssetDatabase for an Asset of type T using the searchFilter string.
+        /// See https://docs.unity3d.com/ScriptReference/AssetDatabase.FindAssets.html for what can be included in that
+        /// string.
+        /// </summary>
+        /// <returns>The first found asset or null if no assets are found.</returns>
+        public static T FindAssetByFilter<T>(string searchFilter) where T : UnityEngine.Object
         {
-            // We search for the sprite by name instead of by its path, because we don't require developers to keep the
+            // We search for the asset by name instead of by its path, because we don't require developers to keep the
             // play-instant-unity-plugin folder directly inside the Assets folder.
-            string searchFilter = "GooglePlayInstantRetryButton t:sprite";
             string[] foundGuids = AssetDatabase.FindAssets(searchFilter);
 
             if (foundGuids.Length <= 0)
```

Comment:
Nit: "== 0" is sufficient.
Should there also be an error if Length > 1 ?

Reply:
That's a good point. I'll add an error if we find more than one element.

Reply:
If there is more than one element, does it make more sense to throw an error and return null, or throw an error and return the first one?